EKF/control.cpp

```diff
@@ -977,6 +977,18 @@ void Ekf::controlMagFusion()
 				_control_status.flags.mag_hdg = true;
 			}
 
+			// perform switch-over from only updating the mag states to updating all states
+			if (!_control_status.flags.update_mag_states_only && _control_status_prev.flags.update_mag_states_only) {
+				// reset the mag field covariances
+				zeroRows(P, 16, 21);
+				zeroCols(P, 16, 21);
+
+				// re-instate the last used variances
+				for (uint8_t index = 0; index <= 5; index ++) {
+					P[index+16][index+16] = _saved_mag_variance[index];
```

Comment:
After further thought I think that when we come out of the inhibit condition, if we are still in 3-axis fusion, then the variances should be set to alignment values.